isobus/src/can_message.cpp

```diff
@@ -62,4 +62,97 @@ namespace isobus
 		return CANPortIndex;
 	}
 
+	std::uint8_t CANMessage::get_uint8_at(const std::size_t index)
+	{
+		std::uint8_t retVal = 0;
```

Comment:
Since the default return values of 0 could be "valid" or legal values, do you think it might make sense to assert on the index boundary check? I think assertion would be fairly reasonable to do, as trying to access a value without checking the message DLC is more of a development error? Or, maybe instead it makes sense to make these always do `.at` so that the error gets caught more easily?

Reply:
I see, that makes sense. I'll change it

Reply:
Fixed in https://github.com/ad3154/ISO11783-CAN-Stack/pull/138/commits/0cd2603d4f9f2f75d37433d11e85ecb4f59ac04f. I chose for the `vector::at()` as asserts typically get removed with "release" builds, and with the stack still in somewhat early development, I feel like it will be helpful someday to not have the boundary check optimized out for now. We can change this later when optimizations are needed I guess.